Enterprise GeoIp Task experiment

masseyke opened the pull request:
DRAFT

masseyke committed:
Enterprise GeoIp Task experiment

masseyke committed:
deleted junk, spotlessapply, added transport version

masseyke committed:
LicensedEnterpriseGeoIpDownloaderTaskExecutor does not need to be a PersistentTaskExecutor

masseyke committed:
renamed LicensedEnterpriseGeoIpDownloaderTaskExecutor to LicensedEnterpriseGeoIpDownloaderTaskExecutor

joegallo commented on modules/ingest-geoip/src/main/java/module-info.java:
```diff
@@ -16,4 +16,5 @@
     requires com.maxmind.db;
 
     exports org.elasticsearch.ingest.geoip.stats to org.elasticsearch.server;
+    exports org.elasticsearch.ingest.geoip.enterprise;
```
Is there some `to [...]` that could go here that would work?

joegallo committed:
Simplify slightly, treeshaking

joegallo committed:
TODOs as review

joegallo committed:
WIP switching to a LicenseStateListener

joegallo committed:
Add a module for the enterprise downloader

this lets us tighten the binding on the export side to say that we're
only exporting this weird stuff into the enterprise downloader, which
feels a little like a tidy wormhole for us to ship code through.

joegallo committed:
Reorder slightly to decrease overall diff

joegallo committed:
Whitespace

joegallo committed:
More treeshaking

joegallo committed:
Add a todo and the state to the logs

joegallo committed:
Merge branch 'main' into enterprise-geoip-task-experiment

joegallo committed:
A slightly simpler approach

This is a little more direct, but it requires that we put the code
that needs to be on both sides of the aisle into server, rather than
exporting code from the ingest-geoip module out into the license-aware
side of the house.

I'm keeping this here just as a WIP commit to give us a view of the
two sides at the same time -- we really could do it either way.
